src/features/user/sagas.ts

```diff
@@ -1,20 +1,34 @@
 import { AxiosError } from 'axios'
-import { all, fork, put, takeLatest } from 'redux-saga/effects'
+import { all, call, delay, fork, put, takeLatest } from 'redux-saga/effects'
 
 import { toast } from '../../components/Toaster/toast'
+import { store } from '../../context/store'
 import userService, {
   UserCredential,
   UserDeletionCredential,
 } from '../../services/userService'
 import { Action, CommonSagaActions } from '../../utils/types'
 import { addLoadingTask, removeLoadingTask } from '../common/slice'
+import { getIsLoggedIn } from './selector'
 import { setIsLoggedIn, setUserProfile } from './slice'
 import { UserSagaActions, cookieIsLoggedInKey } from './types'
 
-function* initIsLoggedIn() {
-  yield put(
-    setIsLoggedIn(document.cookie.includes(`${cookieIsLoggedInKey}=true`)),
-  )
+const syncIsLoggedInDelayMillis: number = 1000
+const validateSessionDelayMillis: number = 30000
+
+function* periodicallySyncIsLoggedInCookie() {
+  while (true) {
+    const currIsLoggedIn: boolean = getIsLoggedIn(store.getState())
+    const correctIsLoggedIn: boolean = document.cookie.includes(
+      `${cookieIsLoggedInKey}=true`,
+    )
+
+    if (currIsLoggedIn !== correctIsLoggedIn) {
+      yield put(setIsLoggedIn(correctIsLoggedIn))
+    }
+
+    yield delay(syncIsLoggedInDelayMillis)
```

Comment:
Might be better to convert this to use redux saga's yield race
https://redux-saga.js.org/docs/advanced/RacingEffects